### PR TITLE
WIP: Remove locales

### LINKED ISSUE #40: Huge increase in fullOptJS file size
With version 2.0, we see a huge increase in the js file size. Before the update to `scribe` our bundled size in fullOptJS was `1.4mb`, now with the new version of scribe it is `2.8mb`.

I suspect this is due to the usage of `scala-java-time` and `locales`. We had similar issue in our own app, which is why we have removed these two packages from our app. Now, those are pulled in by scribe. Can we make this optional?

### PR DESCRIPTION
**Pull request by fdietze**
This is an early approach to remove the locales dependency for js builds.

In our project it removed ~2MB from `fastOpt` and ~600kb from `fullOpt` assets.

It's only about `Double` and `Long` formatting which should be possible to implement in pure Scala(JS).

fixes https://github.com/outr/scribe/issues/40
@cornerman

### CODE REVIEW

File: core/js/src/main/scala/perfolation/DoubleImplicits.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class DoubleImplicits(val d: Double) extends AnyVal {
+  def f(i: Int = 1,
+        f: Int = 2,
+        maxI: Int = 9,
+        maxF: Int = 100,
+        g: Boolean = true,
+        c: Option[Currency] = None,
+        rm: RoundingMode = RoundingMode.HALF_UP): String = {
+    d.toString
+//    ThreadLocalNumberFormat(i, f, maxI, maxF, g, c, rm).format(d)
```
Comment (darkfrog26) [inline core/js/src/main/scala/perfolation/DoubleImplicits.scala:15]: I hope you realize I can't merge this until this is fixed, right?

Comment (cornerman) [reply core/js/src/main/scala/perfolation/DoubleImplicits.scala:15]: Of course not :) It was more of an experiment to see what the difference in size is. For us, the problem is that we do not really need such sophisticated formatters and it is totally not worth 600 kB in our javascript asset. Logging in the frontend is mostly just for development, in production we only log exceptional problems to our backend.

Alternatively, could we make this formatting somehow optional? Maybe in a separate subproject in scribe that I can depend on?

Comment (darkfrog26) [reply core/js/src/main/scala/perfolation/DoubleImplicits.scala:15]: Perhaps, but I think we should just figure out a pure Scala way to accomplish this without using Java classes at all and everyone will be happy.

**Comment by cornerman** [comment]
Nice, I think this is done with https://github.com/outr/perfolation/commit/9ad70046b6283d56b09a3b56a484b3441d5d0922 now :)

**Comment by darkfrog26** [comment]
Agreed, we should have a new release this week that will drastically improve support for Scala.js and Native.

**Comment by ebruchez** [comment]
@darkfrog26 was there ever a new release addressing this?

**Comment by darkfrog26** [comment]
Sorry about that. We were working on the 1.1 release and got sidetracked on other tasks. Let me get back to you.  I'll try to do a maintenance release ASAP.

**Comment by ebruchez** [comment]
Thanks, that would be great!

**Comment by darkfrog26** [comment]
Perfolation 1.1.0 is released with this fix.

**Comment by ebruchez** [comment]
Amazing, thanks!

**Pull request closed without merging**